parquet-avro/src/main/java/org/apache/parquet/avro/AvroParquetReader.java

```diff
@@ -66,4 +69,47 @@ public AvroParquetReader(Configuration conf, Path file) throws IOException {
   public AvroParquetReader(Configuration conf, Path file, UnboundRecordFilter unboundRecordFilter) throws IOException {
     super(conf, file, new AvroReadSupport<T>(), unboundRecordFilter);
   }
+
+  public static class Builder<T> extends ParquetReader.Builder {
+
+    private GenericData model = null;
+    private boolean enableCompatibility = true;
+    private boolean isReflect = true;
+
+    private Builder(Path path) {
+      super(path);
+    }
+
+    public Builder<T> withDataModel(GenericData model) {
+      this.model = model;
+
+      // only generic and specific are supported by AvroIndexedRecordConverter
+      if (model.getClass() != GenericData.class &&
+          model.getClass() != SpecificData.class) {
+        isReflect = true;
+      }
+
+      return this;
+    }
+
+    public Builder<T> disableCompatibility() {
+      this.enableCompatibility = false;
+      return this;
+    }
+
+    public Builder<T> withCompatibility(boolean enableCompatibility) {
+      this.enableCompatibility = enableCompatibility;
+      return this;
+    }
+
+    @Override
+    protected ReadSupport<T> getReadSupport() {
+      if (isReflect) {
```

Comment:
isReflect is always true. Shouldn't it be false by default?
The else below is unusable because of this.